synapse_data_warehouse/synapse_event/dynamic_tables/V2.52.0__node_event_table.sql

```diff
@@ -0,0 +1,92 @@
+USE SCHEMA {{ database_name }}.SYNAPSE_EVENT;
+
+CREATE OR REPLACE DYNAMIC TABLE node_event
+    TARGET_LAG = '1 day'
+    WAREHOUSE = compute_xsmall
+    AS
+        WITH latest_unique_rows AS (
+            SELECT
+                CHANGE_TYPE,
+                CHANGE_TIMESTAMP,
+                CHANGE_USER_ID,
+                SNAPSHOT_TIMESTAMP,
+                ID,
+                BENEFACTOR_ID,
+                PROJECT_ID,
+                PARENT_ID,
+                NODE_TYPE,
+                CREATED_ON,
+                CREATED_BY,
+                MODIFIED_ON,
+                MODIFIED_BY,
+                VERSION_NUMBER,
+                FILE_HANDLE_ID,
+                NAME,
+                IS_PUBLIC,
+                IS_CONTROLLED,
+                IS_RESTRICTED,
+                SNAPSHOT_DATE,
+                EFFECTIVE_ARS,
+                ANNOTATIONS,
+                DERIVED_ANNOTATIONS,
+                VERSION_COMMENT,
+                VERSION_LABEL,
+                ALIAS,
+                ACTIVITY_ID,
+                COLUMN_MODEL_IDS,
+                SCOPE_IDS,
+                ITEMS,
+                REFERENCE,
+                IS_SEARCH_ENABLED,
+                DEFINING_SQL,
+                INTERNAL_ANNOTATIONS,
+                VERSION_HISTORY,
+                PROJECT_STORAGE_USAGE
+            FROM
+                {{ database_name }}.synapse_raw.nodesnapshots --noqa: TMP
+            WHERE
+                snapshot_timestamp >= CURRENT_TIMESTAMP - INTERVAL '30 DAYS'
+            QUALIFY ROW_NUMBER() OVER (
+                    PARTITION BY id, version_number, change_type
```

Comment:
You are able to "update" the same version over and over.  Maybe look at etag?

Reply:
I think this answers your concerns? https://github.com/Sage-Bionetworks/snowflake/pull/205#discussion_r2093382761

Reply:
Addressed by [d9fc4b6](https://github.com/Sage-Bionetworks/snowflake/pull/205/commits/d9fc4b677cccbaf0c023693a20a451e8076aac94).